File: RelatedWordsAPI/Models/Project.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace RelatedWordsAPI.Models
{
    public enum ProjectProcessingStatus
    {
        NotStarted,
        Processing,
        Finished,
        Canceled,
        Failed,
        Unknown,
    }

    public class Project
    {
        public Project()
        {
            Words = new HashSet<Word>();
        }
        public int ProjectId { get; set; }
        public int UserId { get; set; }
        // Increased, when something changes in the project. May serve to refresh UI.
        public int EditRevisionNumber { get; set; }
        // Increased, when something changes in the project pages. 
        // Together with ProcessedPagesRevisionNumber may serve to know that the processing results are outdated.
        public int EditPagesRevisionNumber { get; set; }
        // Increased, when the project is processed.
        public int ProcessingRevisionNumber { get; set; }
        // Contains the EditPagesRevisionNumber, when the project was last processed.
        public int ProcessedPagesRevisionNumber { get; set; }
        public string Name { get; set; }
        public User User { get; set; }
        public ProjectProcessingStatus ProcessingStatus { get; set; }
        public DateTime CreatedDate { get; set; }
        public ICollection<Page> Pages { get; set; }
        public ISet<Word> Words { get; private set; }
    }
}

File: RelatedWordsUI/related-words-ui/src/models/Project.js
import isDefinedOrNull from './helpers/isDefinedOrNull';

class Project {
  constructor(props) {
    this.projectId = isDefinedOrNull(props.projectId);
    this.userId = isDefinedOrNull(props.userId);
    this.name = isDefinedOrNull(props.name);
    this.processingStatus = isDefinedOrNull(props.processingStatus);
    this.createdDate = isDefinedOrNull(props.createdDate);
    this.editRevisionNumber = isDefinedOrNull(props.editRevisionNumber);
    this.editPagesRevisionNumber = isDefinedOrNull(props.editPagesRevisionNumber);
    this.processingRevisionNumber = isDefinedOrNull(props.processingRevisionNumber);
    this.processedPagesRevisionNumber = isDefinedOrNull(props.processedPagesRevisionNumber);
  }

  processingStatusToString() {
    switch(this.processingStatus) {
      case 0:
        return 'Not Started';
      case 1:
        return 'Processing';
      case 2:
        return 'Finished';
      case 3:
        return 'Canceled';
      case 4:
        return 'Failed';
      case 5:
        return 'Unknown';
      default:
        return 'Unknown';
    }
  }
}

export default Project;